src/track/beatgrid.cpp

```diff
@@ -54,21 +55,20 @@ BeatGrid::BeatGrid(
           m_sampleRate(sampleRate),
           m_grid(grid),
           m_beatLengthFrames(beatLengthFrames) {
-    // BeatGrid should live in the same thread as the track it is associated
-    // with.
 }
 
-BeatGrid::BeatGrid(const BeatGrid& other,
+BeatGrid::BeatGrid(
+        MakeSharedTag,
+        const BeatGrid& other,
         const mixxx::track::io::BeatGrid& grid,
         audio::FrameDiff_t beatLengthFrames)
-        : m_subVersion(other.m_subVersion),
-          m_sampleRate(other.m_sampleRate),
-          m_grid(grid),
-          m_beatLengthFrames(beatLengthFrames) {
+        : BeatGrid({}, other.m_sampleRate, other.m_subVersion, grid, beatLengthFrames) {
 }
 
-BeatGrid::BeatGrid(const BeatGrid& other)
-        : BeatGrid(other, other.m_grid, other.m_beatLengthFrames) {
+BeatGrid::BeatGrid(
+        MakeSharedTag,
+        const BeatGrid& other)
+        : BeatGrid({}, other, other.m_grid, other.m_beatLengthFrames) {
```

Comment:
Why you don't pass the MakeSharedTag, but create a new one?

Reply:
Because type tags are not supposed to carry any information and should be eliminated during compilation. Passing the instance is not desired in this special case. Just a hint for the compiler to ignore the unnamed parameter.